src/write_ext.rs

```diff
@@ -17,9 +17,8 @@ pub trait WriteBinProtExt: io::Write {
     }
 
     // chars are 1 byte long
-    fn bin_write_char(&mut self, c: char) -> Result<usize, io::Error> {
-        self.write_u8(c as u8)?;
-        Ok(1)
+    fn bin_write_char(&mut self, c: char) -> Result<(), io::Error> {
```

Comment:
We should switch back to the older return type with a usize as we want the information of how many bytes were written as all Writer APIs in general are expected to return number of bytes written.

Reply:
Done